delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/resources/ResourceEvent.scala

```diff
@@ -0,0 +1,170 @@
+    * @return the resource identifier
+    */
+  def id: Iri
+
+  /**
+    * @return the project where the resource belongs to
+    */
+  def project: ProjectRef
+
+  /**
+    * @return the collection of known resource types
+    */
+  def types: Set[Iri]
+
+}
+
+object ResourceEvent {
+
+  /**
+    * Event representing a resource creation.
+    *
+    * @param id          the resource identifier
+    * @param project     the project where the resource belongs
+    * @param schema      the schema used to constrain the resource
+    * @param types       the collection of known resource types
+    * @param source      the representation of the resource as posted by the subject
+    * @param compacted   the compacted JSON-LD representation of the resource
+    * @param expanded    the expanded JSON-LD representation of the resource
+    * @param rev         the resource revision
+    * @param instant     the instant when this event was created
+    * @param subject     the subject which created this event
+    */
+  final case class ResourceCreated(
+      id: Iri,
+      project: ProjectRef,
+      schema: ResourceRef,
+      types: Set[Iri],
+      source: Json,
+      compacted: CompactedJsonLd,
+      expanded: ExpandedJsonLd,
+      rev: Long,
+      instant: Instant,
+      subject: Subject
+  ) extends ResourceEvent
+
+  /**
+    * Event representing a resource modification.
+    *
+    * @param id          the resource identifier
+    * @param project     the project where the resource belongs
+    * @param types       the collection of known resource types
+    * @param source      the representation of the resource as posted by the subject
+    * @param compacted   the compacted JSON-LD representation of the resource
+    * @param expanded    the expanded JSON-LD representation of the resource
+    * @param rev         the resource revision
+    * @param instant     the instant when this event was created
+    * @param subject     the subject which created this event
+    */
+  final case class ResourceUpdated(
+      id: Iri,
+      project: ProjectRef,
+      types: Set[Iri],
+      source: Json,
+      compacted: CompactedJsonLd,
+      expanded: ExpandedJsonLd,
+      rev: Long,
+      instant: Instant,
+      subject: Subject
+  ) extends ResourceEvent
+
+  /**
+    * Event representing a tag addition to a resource.
+    *
+    * @param id        the resource identifier
+    * @param project   the project where the resource belongs
+    * @param types     the collection of known resource types
+    * @param targetRev the revision that is being aliased with the provided ''tag''
+    * @param tag       the tag of the alias for the provided ''targetRev''
+    * @param rev       the resource revision
+    * @param instant   the instant when this event was created
+    * @param subject   the subject which created this event
+    */
+  final case class ResourceTagAdded(
+      id: Iri,
+      project: ProjectRef,
+      types: Set[Iri],
```

Comment:
We don't need the types as they don't change and are not a part of the resource identifier no ?

Reply:
The types are some information we want to carry on the event. This is for indexing purposes, since some indexing decisions will depend on the types available

Reply:
Also for backwards compatibility of the SSE.